pontoon/contributors/views.py

```diff
@@ -411,13 +423,122 @@ def settings(request):
             "locale": custom_homepage_locale,
             "preferred_locales": preferred_locales,
             "preferred_locale": preferred_source_locale,
+            "personal_access_tokens": personal_access_tokens,
+            "now": timezone.now(),
             "user_form": user_form,
             "user_profile_form": user_profile_form,
             "user_profile_toggle_form": forms.UserProfileToggleForm(instance=profile),
         },
     )
 
 
+@login_required(redirect_field_name="", login_url="/403")
+@require_AJAX
+def generate_token(request):
+    # generates a 43 char token with characters a-z, A-Z, 0-9
+    def generate_unhashed_token():
+        return "".join(
+            secrets.choice(string.ascii_letters + string.digits) for _ in range(43)
+        )
+
+    try:
+        name = request.POST["name"]
+    except KeyError:
+        return JsonResponse(
+            {"status": "error", "message": "Token name is required."}, status=400
+        )
+    if request.method == "POST":
+        pat_count = PersonalAccessToken.objects.filter(user=request.user).count()
+
+        if pat_count < PERSONAL_ACCESS_TOKEN_MAX_COUNT:
```

Comment:
You can save a level of indentation by flipping the condition. If the count is 10 or more, you return the error message. If not, you just continue after the `if` block, no need for an `else`.